### ISSUE TITLE
JJ seems to ignore configuration on MacOS

### ISSUE THREAD

**Issue by fride** [martinvonz/jj#4406]
## Description

Apparently JJ seems to ignore my user configuration on my machine. When I checkout a project (like eg this one) it warns me about missing user name and email, even though both are configured in `~/Library/Application Support/<username>/jj/config.toml`

## Steps to Reproduce the Problem

1. configure user name and email
2. Clone a git repository: `jj git clone https://github.com/martinvonz/jj.git`
3. There are warning about missing configurations

## Expected Behavior

There should be no warnings.

## Actual Behavior

JJ claims that neither user name nor email are set.
```
Fetching into new repo in "/private/tmp/jj"
..
Warning: Name and email not configured. Until configured, your commits will be created with the empty identity, and can't be pushed to remotes. To configure, run:
  jj config set --user user.name "Some One"
  jj config set --user user.email "someone@example.com"
...

```
## Specifications

- Platform: MacOS 14.16.1 Arm.
- Version: 0.20.0

**Comment by martinvonz** [comment]
What does `jj config path --user` say? Maybe jj thinks the config is in a different file than `~/Library/Application Support/<username>/jj/config.toml`.

**Comment by fride** [comment]
```
❯ jj config path --user
/Users/jgf/Library/Application Support/jj/config.toml

**Comment by fride** [comment]
It says the expected
```
jj config path --user
/Users/<me>/Library/Application Support/jj/config.toml
```

I copied the config to `~/.config/jj/config.toml` and had the same result.

**Comment by martinvonz** [comment]
And `jj config list --user user` lists your configured values? Here's my outout for reference:

```shell
$ jj config list --user user
user.email = "martinvonz@google.com"
user.name = "Martin von Zweigbergk"
```

**Comment by fride** [comment]
Yes, everything is there. Maybe the git configuration is in a place that jj dislikes?

**Comment by fride** [comment]
OMG! No. Thanks a lot. 🙈

**Comment by fride** [comment]
Typo on my side. Sorry for the inconvenience.

**Comment by martinvonz** [comment]
We could make the error message more specific and say which of the two (name or email) are missing. Maybe that would have helped you spot the problem more quickly.

**Comment by fride** [comment]
As my scatterbrain produced it I could try this as my first PR. Do you think that would be 1) easy and 2) a useful addition?

**Comment by martinvonz** [comment]
Yes, should be very easy. A more specific message seems like it would have been useful to at least you :) I'm not sure exactly how specific we should make it in the three different scenarios. I'll let you figure that out.

**Comment by lukerandall** [comment]
Fixed in #4429
